lib/Asm89/Twig/CacheExtension/Node/CacheNode.php

```diff
@@ -47,6 +68,15 @@ public function compile(\Twig_Compiler $compiler)
                 ->raw(", ")
                 ->subcompile($this->getNode('key_info'))
             ->write(");\n")
+            ->write("if (isset(\$asm89Key".$i."['key']['key'])) {\n")
+                ->indent()
+                    ->write("\$asm89Key".$i."['key']['key'] .= '/".$hash."';\n")
+                ->outdent()
+            ->write("} elseif (isset(\$asm89Key".$i."['key'])) {\n")
+                ->indent()
+                    ->write("\$asm89Key".$i."['key'] .= '/".$hash."';\n")
+                ->outdent()
+            ->write("}\n")
```

Comment:
Wondering if it would make sense to do this some other way (pass the hash to `generateKey()` or `fetchBlock()`, wrap the cache strategy in something that also takes the hash).